crates/rspack_core/src/concatenated_module.rs

```diff
@@ -1989,10 +1989,41 @@ impl ConcatenatedModule {
       let comments = SwcComments::default();
       let mut module_info = concatenation_scope.current_module;
 
+      let jsx = {
+        let get_jsx = |options: &ParserOptions| -> Option<bool> {
+          match options {
+            ParserOptions::Javascript(js_options)
+            | ParserOptions::JavascriptAuto(js_options)
+            | ParserOptions::JavascriptEsm(js_options)
+            | ParserOptions::JavascriptDynamic(js_options) => js_options.jsx,
+            _ => None,
+          }
+        };
+
+        module
+          .as_ref()
+          .as_normal_module()
+          .and_then(|normal_module| normal_module.get_parser_options())
+          .and_then(get_jsx)
```

Comment:
Use `get_javascript().jsx` here, it's the simple way to handle get options from different js types